sdk/js/src/RequestUtils.ts

```diff
@@ -0,0 +1,159 @@
+) => RequestInitWithRetry['retryOn']
+
+const retryOnRequestError: retryOnRequestErrorFunc = (retries) => {
+    return (attempt, error, response) => {
+        if (attempt >= retries) {
+            return false
+        } else if (response && response?.status < 500) {
+            return false
+        }
+
+        return true
+    }
+}
+
+export async function handleResponse(res: Response): Promise<Response> {
+    // res.ok only checks for 200-299 status codes
+    if (!res.ok && res.status >= 400) {
+        let error
+        try {
+            const response: any = await res.clone().json()
+            error = new ResponseError(
+                response.message || 'Something went wrong',
+            )
+        } catch (e) {
+            error = new ResponseError('Something went wrong')
+        }
+        error.status = res.status
+        throw error
+    }
+
+    return res
+}
+
+export async function getWithTimeout(
+    url: string,
+    requestConfig: RequestInit | RequestInitWithRetry,
+    timeout: number,
+): Promise<Response> {
+    const controller = new AbortController()
+    const id = setTimeout(() => {
+        controller.abort()
+    }, timeout)
+    const response = await get(url, {
+        ...requestConfig,
+        signal: controller.signal,
+    })
+    clearTimeout(id)
+    return response
+}
+
+export async function post(
+    url: string,
+    requestConfig: RequestInit | RequestInitWithRetry,
+    sdkKey: string,
+): Promise<Response> {
+    const [_fetch, config] = await getFetchAndConfig(requestConfig)
+    const postHeaders = {
+        ...config.headers,
+        Authorization: sdkKey,
+        'Content-Type': 'application/json',
+    }
+
+    const res = await _fetch(url, {
+        ...config,
+        headers: postHeaders,
+        method: 'POST',
+    })
+
+    return handleResponse(res)
+}
+
+export async function patch(
+    url: string,
+    requestConfig: RequestInit | RequestInitWithRetry,
+    sdkKey: string,
+): Promise<Response> {
+    const [_fetch, config] = await getFetchAndConfig(requestConfig)
+    const patchHeaders = {
+        ...config.headers,
+        Authorization: sdkKey,
+        'Content-Type': 'application/json',
+    }
+
+    const res = await _fetch(url, {
+        ...config,
+        headers: patchHeaders,
+        method: 'PATCH',
+    })
+
+    return handleResponse(res)
+}
+export async function get(
+    url: string,
+    requestConfig: RequestInit | RequestInitWithRetry,
+): Promise<Response> {
+    const [_fetch, config] = await getFetchAndConfig(requestConfig)
+    const headers = { ...config.headers, 'Content-Type': 'application/json' }
+
+    const res = await _fetch(url, {
+        ...config,
+        headers,
+        method: 'GET',
+    })
+
+    return handleResponse(res)
+}
+
+async function getFetch() {
+    if (typeof fetch !== 'undefined') {
+        return fetch
+    }
+
+    return (await import(/* webpackMode: "eager" */ 'cross-fetch')).default
```

Comment:
why dynamically import this if forcing webpack to always include it anyway?

also why do we actually need cross-fetch again?

Reply:
Because we have more than just a webpack build,
and I believe it's to import fetch for environments without access to it natively

Reply:
what environments is the JS SDK used in that dont have access to fetch?